Core/Newsletter2Go/Subscriber/CookieRegisterer.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace Newsletter2Go\Subscriber;
+
+
+use Enlight\Event\SubscriberInterface;
+use Shopware\Bundle\CookieBundle\CookieCollection;
+use Shopware\Bundle\CookieBundle\Structs\CookieGroupStruct;
+use Shopware\Bundle\CookieBundle\Structs\CookieStruct;
+
+class CookieRegisterer implements SubscriberInterface
+{
+
+  public static function getSubscribedEvents()
+  {
+    return [
+          'CookieCollector_Collect_Cookies' => 'addTrackingCookies',
+        ];
+    }
+
+  /**
+   * @param \Enlight_Event_EventArgs $args
+   */
+  public function addTrackingCookies(\Enlight_Event_EventArgs $args)
+  {
+    try {
+        $collection = new CookieCollection();
+      $collection->add(new CookieStruct(
+        'n2g',
+        '/^n2g/',
```

Comment:
in this regex we have to be more precise like '/^n2g$/' to only match n2g 
source: the example from [here](https://developers.shopware.com/developers-guide/cookie-consent-manager/#registering-your-own-cookie)